lib/inspec/control_eval_context.rb

```diff
@@ -19,11 +19,16 @@ class ControlEvalContext
     #
     # @param [ResourcesDSL] resources_dsl which has all resources to attach
     # @return [RuleContext] the inner context of rules
-    def self.rule_context(resources_dsl)
+    def self.rule_context(resources_dsl, profile_context_owner)
       require 'rspec/core/dsl'
       Class.new(Inspec::Rule) do
         include RSpec::Core::DSL
         with_resource_dsl resources_dsl
+
+        # allow attributes to be accessed within control blocks
+        define_method :attribute do |name, options = {}|
+          profile_context_owner.find_or_create_attribute(name, options)
```

Comment:
Find or create here makes me squeamish.  In a control, is it proper to _create_ an attribute? Shouldn't you only be reading values of already created attributes?